source/postprocess/visualization/dynamic_topography.cc

```diff
@@ -75,21 +77,36 @@ namespace aspect
           if (cell->is_locally_owned())
             if (cell->at_boundary())
               {
-                // see if the cell is at the *top* boundary, not just any boundary
-                unsigned int top_face_idx = numbers::invalid_unsigned_int;
-                {
-                  for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
-                    if (cell->at_boundary(f) && this->get_geometry_model().depth (cell->face(f)->center()) < cell->face(f)->minimum_vertex_distance()/3)
+                // see if the cell is at the *top* or *bottom* boundary, not just any boundary
+                unsigned int face_idx = numbers::invalid_unsigned_int;
+                bool at_upper_surface;
```

Comment:
Additionally, can a cell be on the "upper" and the "lower" surface at the same time? If not, can this be checked with an Assert()?